CoreFoundation/Base.subproj/CFKnownLocations.c

```diff
@@ -96,30 +97,12 @@ CFURLRef _Nullable _CFKnownLocationCreatePreferencesURLForUser(CFKnownLocationUs
         }
         case _kCFKnownLocationUserCurrent:
             username = CFGetUserName();
-            // fallthrough
+            break;
         case _kCFKnownLocationUserByName: {
-            DWORD size = 0;
-            GetProfilesDirectoryW(NULL, &size);
-
-            wchar_t* path = (wchar_t*)malloc(size * sizeof(wchar_t));
-            GetProfilesDirectoryW(path, &size);
-
-            CFStringRef pathRef = CFStringCreateWithCharacters(kCFAllocatorSystemDefault, path, size - 1);
-            free(path);
-
-            CFURLRef profilesDir = CFURLCreateWithFileSystemPath(kCFAllocatorSystemDefault, pathRef, kCFURLWindowsPathStyle, true);
-            CFRelease(pathRef);
-
-            CFURLRef usernameDir = CFURLCreateCopyAppendingPathComponent(kCFAllocatorSystemDefault, profilesDir, username, true);
-            CFURLRef appdataDir = CFURLCreateCopyAppendingPathComponent(kCFAllocatorSystemDefault, usernameDir, CFSTR("AppData"), true);
-            location = CFURLCreateCopyAppendingPathComponent(kCFAllocatorSystemDefault, appdataDir, CFSTR("Local"), true);
-            CFRelease(usernameDir);
-            CFRelease(appdataDir);
-
-            CFRelease(profilesDir);
-            if (user == _kCFKnownLocationUserCurrent) {
-                CFRelease(username);
-            }
+            wchar_t* path = NULL;
+            SHGetKnownFolderPath(&FOLDERID_LocalAppData, 0, NULL, &path);
+            location = CFURLCreateWithFileSystemPath(kCFAllocatorSystemDefault, path, kCFURLWindowsPathStyle, true);
```

Comment:
This seems incorrect. The `path` parameter is a wide C String not a CFStrimg. This should do the creation of the CFString and pass that.